File: week7/log_probability.pl
#!/usr/bin/perl -w
use strict;

my $count = 0;
my $words = 0;
my $target = lc($ARGV[0]);
my %poets = ();
foreach my $poet (glob "poets/*.txt") {
	$count = 0;
	$words = 0;
	open(F,"<$poet") or die "$0: Can't open $poet: $!\n";
	while (<F>) {
		#$words += scalar(split(/[a-zA-Z]/, $_));
		my @items = split(/[^A-Za-z]/, $_);
		my @new;
		foreach(@items){
		    if( ( defined $_) and ($_ =~ /^$target$/i )){
		        $count++;
		    }
		    if( ( defined $_) and !($_ =~ /^$/ )){
		        push(@new, $_);
		    }
		}
		$words += scalar(@new);
	}
	$poet =~ s/poets\///;
	$poet =~ s/\.txt//;
	$poet =~ tr/_/ /;
	$poets{"$poet"}{"$target"} = $count;
	$poets{"$poet"}{total} = $words;
	printf "log((%d+1)/%6d) = %8.4f %s\n", $poets{"$poet"}{"$target"}, $poets{"$poet"}{total}, log(($poets{"$poet"}{"$target"}+1)/$poets{"$poet"}{total}), $poet;
}

File: week7/log_probability.py
#!/usr/bin/python

import sys
import fileinput
import re
import glob
import string
from string import maketrans
import math

for file in glob.glob("poets/*.txt"):
	words = [];
	input = [];
	f = open(file, 'r');
	for lines in f:
		lines = lines.lower()
		input += re.split(r'[^a-zA-Z]', lines);
	words += filter(None, input);
	counter = words.count((sys.argv[1]).lower());
	poet = file;
	m = re.search('(?<=poets/)\w+', file);
	poet = m.group(0)
	poet = poet.replace("_", " ")
	#print "%(counter)4d/%(words)6d = %(frequency).9f %(poet)s" % {"counter": counter, "words": len(words), "frequency": float(counter)/len(words), "poet": poet}
	print "log((%(counter)d+1)/%(words)6d) = %(frequency)8.4f %(poet)s" % {"counter": counter, "words": len(words), "frequency": math.log((float(counter + 1)/len(words))), "poet": poet}